dotnet/src/Connectors/Connectors.Memory.Sqlite/Database.cs

```diff
@@ -168,6 +168,27 @@ public Task DeleteAsync(SqliteConnection conn, string collectionName, string key
         return cmd.ExecuteNonQueryAsync(cancellationToken);
     }
 
+    public Task DeleteBatchAsync(SqliteConnection conn, string collectionName, string[] keys, CancellationToken cancellationToken = default)
+    {
+        using SqliteCommand cmd = conn.CreateCommand();
+        var keyParameters = keys.Select((key, index) => $"@key{index}").ToArray();
```

Comment:
No need to do a `ToArray` here, since string.Join also takes `IEnumerable` as a parameter.  Also, doing a `ToArray` forces an iteration of `keys`, and then string.join does it again, so removing `ToArray`, removes one iteration.

Reply:
Okay let me update the code.

Reply:
@westey-m if I change `string[] keys` to `IEnumerable<string>keys` it is showing CA1851 rule warning

Reply:
I meant you can update line 174 like this:
```cs
        var keyParameters = keys.Select((key, index) => $"@key{index}");
        var parameters = string.Join(", ", keyParameters);
```

Reply:
Oh, thanks. Updated @westey-m

Reply:
Looks good and thanks for the contribution!